src/osm_configurator/view/toplevelframes/calculation_frame.py

```diff
@@ -175,8 +202,9 @@ def __aggregation_pressed(self):
         """
         checker = self._calculation_controller.start_calculations(CalculationPhase.AGGREGATION_PHASE)
         if checker != CalculationPhase.RUNNING:  # Check, if changing states is possible
+            # Interrupt calculation start
             self.__calculation_start_interrupted()
-            self.activate()
+            self.activate()  # TODO: figure out how to reload a page
```

Comment:
yes, your activate() should reload your page

Reply:
as a tip on how to do that, figure out what information your frame presents and where it gets those, so at the beginning you collect all infomation via the controller that you need, basicly reset all your buttons and other elements to a default state and then add the collected information to it